src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,210 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+public class Parser {
+    public static int parse(String text, UI ui, TaskList list, Storage storage) throws DukeException {
+
+
+            if (text.length() > 3 && text.substring(0, 4).equals("list")) {
+                ui.printline();
+
+                for (int i = 0; i < list.size(); i++) {
+                    if (list.get(i) == null) {
+                        break;
+                    } else {
+                        ui.sendMessage(String.format("%d. [%s] [%s] %s", i + 1, list.get(i).tag,
+                                list.get(i).getStatusIcon(), list.get(i)));
+                    }
+                }
+                ui.printline();
+            } else if (text.startsWith("unmark")) {
+                try {
+                    storage.appendToFile(text + "\n");
+                } catch (IOException e) {
+                    ui.sendMessage("Something went wrong: " + e.getMessage());
+                }
+                String[] splitText = text.split(" ");
+                int numToUnmark = Integer.parseInt(splitText[1]) - 1;
+                list.get(numToUnmark).markAsIncomplete();
+
+                ui.printline();
+                ui.sendMessage("Alright! I'll uncheck this task for you: ");
+                ui.sendMessage(String.format("\t [%s] [%s] %s", list.get(numToUnmark).tag,
+                        list.get(numToUnmark).getStatusIcon(), list.get(numToUnmark)));
+                ui.printline();
```

Comment:
You might want to consider extracting all these out and putting it in a function in the Ui class. This can help your Parser class to be easier to understand. Same comment for each of the conditions below. 

For example, in your Ui class you can have a show `UnmarkMessage()` method etc.